nextcloudappstore/core/views.py

```diff
@@ -18,19 +20,48 @@ def get_context_data(self, **kwargs):
 class CategoryAppListView(ListView):
     model = App
     template_name = 'app/list.html'
+    allow_empty = True
 
     def get_queryset(self):
         category_id = self.kwargs['id']
         queryset = super().get_queryset()
+
         if category_id:
-            return queryset.filter(categories__id=category_id)
-        else:
-            return queryset
+            queryset = queryset.filter(categories__id=category_id)
+
+        if self.search_terms:
+            query = None
+
+            for term in self.search_terms:
```

Comment:
I opted for | because of its more forgiving behaviour (you can enter some bad search terms and still find something), but I do see your point.
I have no idea how this works with translations.